### PR TITLE
Addded nullChecked support for nullable list

### PR DESCRIPTION
**Pull request by gopalsabhadiya**
I've added nullChecked for nullable list. (Test is failing due to an issue described below).

Generated source is generating:
```
final nullablelisttarget =
        NullableListTarget(source.list!.map((e) => e).toList());
```

It should be

```
final nullablelisttarget =
        NullableListTarget(source.list!.map(fromSubSource).toList());
```

I'm not sure why `matchingMappingListMethods.isNotEmpty` is returning false. 

If you can help me out, I can make changes and push the commit on this PR.
Also, let me know if there are any other changes related to this.

### CODE REVIEW

File: generator/lib/code_builders/assignment_builder.dart [hunk 1]
```diff
@@ -50,16 +51,31 @@ Expression generateSourceFieldAssignment(SourceAssignment sourceAssignment,
...
-              //.toList()
-              .property('toList')
-              .call([]);
+      if (sourceField.type.isDartCoreList &&
+          sourceField.type.nullabilitySuffix == NullabilitySuffix.question) {
+        sourceFieldAssignment =
+            // source.{field}.map
+            sourceReference
+                .property(sourceField.name)
+                .nullChecked
```
Comment (gopalsabhadiya) [inline generator/lib/code_builders/assignment_builder.dart:60]: This has been added to add `!`. Due to this, I had to pub upgrade the dependencies.

**Comment by smotastic** [comment]
Hey @gopalsabhadiya 
thank you so much for your contribution. I took a look at it, and it looks good so far.
I made a few comments in the review section, if you have any questions feel free to ask. 
I think I also found your problem with the failing test, see the review

**Comment by gopalsabhadiya** [comment]
Hi @smotastic 

I failed to find any review comments. Can you please refer me the link to that?

**Review by smotastic — CHANGES_REQUESTED** [review]

**Comment by smotastic** [comment]
Hi @gopalsabhadiya 
Yes, sorry. I forgot to click the "request changes" button. Should be there now above my comment

**Comment by smotastic** [comment]
Hi, I will close this PR for now due to inactivity.
But feel free to open another PR whenever you find the time.

**Pull request closed without merging**